src/geometry/detail/VGNavCollection.cc

```diff
@@ -0,0 +1,97 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2021 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file VGNavCollection.cc
+//---------------------------------------------------------------------------//
+#include "VGNavCollection.hh"
+
+#include <VecGeom/navigation/NavStatePool.h>
+#include "comm/Device.hh"
+
+namespace celeritas
+{
+namespace detail
+{
+//---------------------------------------------------------------------------//
+// HOST VALUE
+//---------------------------------------------------------------------------//
+void VGNavCollection<Ownership::value, MemSpace::host>::resize(int max_depth,
+                                                               size_type size)
+{
+    CELER_EXPECT(max_depth > 0);
+    CELER_EXPECT(size == 1);
```

Comment:
This is surprising ... Is it documented why this particular collection can only be of size 1?

Reply:
I can add to the docs -- but basically it looks like the VecGeom navstatecollection always allocates GPU memory when compiling with CUDA, so we can't use it for cuda-disabled tests (and prob don't want to anyway for efficiency). Since we're not currently running more than one simultaneous track at a time on CPU (no good reason to since we don't have threads) the limitation to 1 should be OK for now right?

Reply:
Unless we are certain that we won't extent it, it would make more sense to prepare it to be `size>1` ready while the implementation is fresh in our memory ... humm actually nevermind, it looks like the implementation is small enough that it does not matter.